### PR TITLE
Use PickleError base class to detect all pickle errors

### LINKED ISSUE #6902: Pickle Error in clean_namespace
## 🐛 Bug

I am using a special model which includes ScriptModules. When I call `trainer.fit()` the `clean_namespace()` function in the training loop throws a `pickle.PickleError`. By my understanding, this function is supposed to remove unpickleable attributes. `is_picklable()` catches `pickle.PicklingError` but not `pickle.PickleError` and thus raises the error all the way up to the user instead of catching it and removing the attribute from the `hparams_dict`. 

### To Reproduce

Use a model with parts that are unpicklable due to being `ScriptModules`. `pickle.PickleError` will be thrown on training start and error out the entire training instead of being caught in `is_picklable()`.

### Expected behavior

Remove the model from `hparams_dict` as I think is intended since the model is not picklable.

### Environment

 - PyTorch Version (e.g., 1.0): 1.7
 - OS (e.g., Linux): Linux
 - How you installed PyTorch (`conda`, `pip`, source): conda
 - Python version: 3.8.8
 - CUDA/cuDNN version: 11.0
 - Any other relevant information: Pytorch Lightning: 1.2.3

Thanks!!

### PR DESCRIPTION
**Pull request by SpontaneousDuck**
## What does this PR do?

Fixes #6902 

## Before submitting
- [x] Was this discussed/approved via a GitHub issue? (not for typos and docs)
- [x] Did you read the [contributor guideline](https://github.com/PyTorchLightning/pytorch-lightning/blob/master/.github/CONTRIBUTING.md), **Pull Request** section?
- [x] Did you make sure your PR does only one thing, instead of bundling different changes together?
- [x] Did you make sure to update the documentation with your changes? (if necessary)
- [x] Did you write any new necessary tests? (not for typos and docs)
- [x] Did you verify new and existing tests pass locally with your changes?
- [x] Did you update the [CHANGELOG](https://github.com/PyTorchLightning/pytorch-lightning/blob/master/CHANGELOG.md)? (not for typos, docs, test updates, or internal minor changes/refactorings)

## PR review
Anyone in the community is free to review the PR once the tests have passed.
Before you start reviewing make sure you have read [Review guidelines](https://github.com/PyTorchLightning/pytorch-lightning/wiki/Review-guidelines). In short, see the following bullet-list:

 - [x] Is this pull request ready for review? (if not, please submit in draft mode)
 - [x] Check that all items from **Before submitting** are resolved
 - [x] Make sure the title is self-explanatory and the description concisely explains the PR
 - [x] Add labels and milestones (and optionally projects) to the PR so it can be classified

## Did you have fun?
First PR!

### CODE REVIEW

**Comment by awaelchli** [comment]
Could you also add a test under tests/utilities/test_parsing.py in the test called "test_is_picklable" with your object that wasn't picklable?

**Comment by SpontaneousDuck** [comment]
How does that look? My model has PyTorch `ScriptModule` in it so I added an empty `ScriptModule` to the test. I verified this throws the same error I am receiving in #6902 .

**Review by awaelchli — APPROVED** [review]
perfect, thanks for the fix

**Review by ananthsub — APPROVED** [review]

**Review by akihironitta — APPROVED** [review]

**Review by tchaton — APPROVED** [review]
Great addition !
